svir/dialogs/load_csv_as_layer_dialog.py

```diff
@@ -69,11 +70,31 @@ def load_from_csv(self):
         else:
             save_dest = None  # the destination file will be selected via GUI
             save_format = None
-        csv_path = self.path
+        if os.path.splitext(self.path)[1] == '.zip':
+            save_dest_dir = tempfile.mkdtemp()
+            # unzip and load all
+            with zipfile.ZipFile(self.path, 'r') as zip_ref:
+                zip_ref.extractall(save_dest_dir)
+            for i, csv_path in enumerate(os.listdir(save_dest_dir)):
+                full_csv_path = os.path.join(save_dest_dir, csv_path)
+                if os.path.isfile(full_csv_path):
+                    if i == 0:
```

Comment:
Since this is only to be used for the statistics, I would search for 'mean' in csv_path and extract that file.
If missing, raise an error. Taking the first file is very dangerous because the order of the files returned by `os.listdir` is unspecified.

Reply:
My idea was to load all the csv files (for all statistics) and to visualize the attribute table of only one, while the others can be visualized by the user afterwards by selecting the corresponding layer. Instead of displaying the table for the first csv, I can look for 'mean' and automatically display that, while still loading also the others as layers.